### PR TITLE
Added a new error

### PR DESCRIPTION
**Pull request by 4gust**
Added a new error for account transfer, "server_error" - it is returned when the request could not be completed on server.

## Proposed changes

Added a new error : `server_error` and subError : `server_error` - this is returned when the server cannot complete the request. 

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/MSIDError.m [hunk 1]
```diff
@@ -124,6 +124,10 @@ MSIDErrorCode MSIDErrorCodeForOAuthErrorWithSubErrorCode(NSString *oauthError, M
     {   //Account Transfter, when user skips the QR code page.
         return MSIDErrorUserCancel;
     }
+    if (oauthError && [oauthError caseInsensitiveCompare:@"server_error"] == NSOrderedSame && [subError caseInsensitiveCompare:@"server_error"] == NSOrderedSame)
+    {   //Account Transfter, when the request could not be completed.
```
Comment (oldalton) [inline IdentityCore/src/MSIDError.m:128]: Are you sure that it makes sense to translate server_error to user cancel error? It seems not very intuitive to me.

File: IdentityCore/src/MSIDError.m [hunk 2]
```diff
@@ -124,6 +124,10 @@ MSIDErrorCode MSIDErrorCodeForOAuthErrorWithSubErrorCode(NSString *oauthError, M
     {   //Account Transfter, when user skips the QR code page.
         return MSIDErrorUserCancel;
     }
+    if (oauthError && [oauthError caseInsensitiveCompare:@"server_error"] == NSOrderedSame && [subError caseInsensitiveCompare:@"server_error"] == NSOrderedSame)
```
Comment (oldalton) [inline IdentityCore/src/MSIDError.m:127]: Should it be handled inside MSIDErrorCodeForOAuthError?

File: IdentityCore/src/MSIDError.h [hunk 3]
```diff
@@ -139,6 +139,7 @@ typedef NS_ENUM(NSInteger, MSIDErrorCode)
     MSIDErrorServerUnauthorizedClient   = -51414,
     MSIDErrorServerDeclinedScopes       = -51415,
     MSIDErrorServerAccessDenied         = -51416,
+    MSIDErrorServerError                = -51417,
```
Comment (oldalton) [inline IdentityCore/src/MSIDError.h:142]: If you're introducing a new top level error here, MSAL needs to be updated accordingly as well, and mapping between MSID and MSAL

File: IdentityCore/src/MSIDError.m (hunk 2 above)
Comment (4gust) [reply IdentityCore/src/MSIDError.m:127]: I understand why we can handle this error in MSIDErrorCodeForOAuthError

File: IdentityCore/src/MSIDError.h (hunk 3 above)
Comment (4gust) [reply IdentityCore/src/MSIDError.h:142]: Thank you for reminding me about this, I will update this when we update the I will update the submodule into the MSAL.

**Pull request merged**